test/e2e/mirror_test.go

```diff
@@ -0,0 +1,152 @@
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/windows-machine-config-operator/controllers"
+	"github.com/openshift/windows-machine-config-operator/pkg/windows"
+)
+
+const (
+	testMirrorSetName               = "e2e-powershell-mirror"
+	upstreamPowershellImageLocation = "mcr.microsoft.com/powershell"
+	badPowershellImageLocation      = "test.registry.io/powershell"
+)
+
+// numBaseRegistryConfigFiles represents the number registry config files that exist before this test suite's steps run
+var numBaseRegistryConfigFiles int
+
+func testImageMirroring(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+
+	require.NoError(t, tc.loadExistingNodes(), "error getting the current Windows nodes in the cluster")
+
+	require.NoError(t, tc.setNumBaseRegistryConfigFiles(), "error getting base number of registry config files")
+
+	if !t.Run("Mirror settings applied to nodes", tc.testMirrorSettingsApplied) {
+		// No point in running the rest of the tests if settings aren't applied
+		return
+	}
+	t.Run("Mirror settings cleared from nodes", tc.testMirrorSettingsCleared)
+}
+
+// setNumBaseRegistryConfigFiles updates the numBaseRegistryConfigFiles variable
+func (tc *testContext) setNumBaseRegistryConfigFiles() error {
+	winNodes := gc.allNodes()
+	if len(winNodes) == 0 {
+		return fmt.Errorf("test requires at least one Windows node to run")
+	}
+	addr, err := controllers.GetAddress(winNodes[0].Status.Addresses)
+	if err != nil {
+		return fmt.Errorf("unable to get node address: %w", err)
+	}
+	numBaseRegistryConfigFiles, err = tc.countItemsInDir(windows.ContainerdConfigDir, addr)
+	if err != nil {
+		return fmt.Errorf("error counting items in dir %s on node %s: %w", windows.ContainerdConfigDir, addr, err)
+	}
+	return nil
+}
+
+// testMirrorSettingsApplied tests that the containerd registry configuration directory is populated
+func (tc *testContext) testMirrorSettingsApplied(t *testing.T) {
+	// enables this test suite to run independently of the creation tests
+	itms, err := tc.ensureMirrorSetExists()
+	require.NoErrorf(t, err, "error ensuring mirror set %s exists", testMirrorSetName)
+	// expected one config file per source entry
+	expectedNumConfigFiles := len(itms.Spec.ImageTagMirrors) + numBaseRegistryConfigFiles
+
+	for _, node := range gc.allNodes() {
+		t.Run(node.Name, func(t *testing.T) {
+			addr, err := controllers.GetAddress(node.Status.Addresses)
+			require.NoError(t, err, "unable to get node address")
+
+			// retry to give time for registry controller to transfer generated config to all nodes
+			err = wait.PollUntilContextTimeout(context.TODO(), 15*time.Second, 5*time.Minute, true,
+				func(ctx context.Context) (done bool, err error) {
+					count, err := tc.countItemsInDir(windows.ContainerdConfigDir, addr)
+					require.NoErrorf(t, err, "error counting items in dir %s on node %s", windows.ContainerdConfigDir, addr)
```

Comment:
As you are already going node by node, I think it would be valuable to add a check that the `test.registry.io/hosts.toml` file has the expected contents on the VM.

Could just be a file read and assert `strings.Contains(out, expectedContents)`

Reply:
This to me is something covered by the `TestGenerateConfig` unit tests. 

The real e2e test validating something like this would be making containerd to consume the host file i.e. [deploying a mirrored workload](https://github.com/openshift/windows-machine-config-operator/compare/8e74444c13342a529cd7da9f97ce3165ba484399..ca7d85ebb17653a7d8f751aa58662de3b59f09bc#diff-099e3332f0e1e5fc59c8cf99775065fdc4b594e7b949b39e3873de36269e0722L40), but we decided this was redundant